binary_search_tree/tree.py

```diff
@@ -14,36 +14,135 @@ class Tree:
+    def inorder_helper(self, current_node, node_list = None):
+        if node_list == None:
+            node_list = []
+        if current_node:
+            self.inorder_helper(current_node.left, node_list)
+            node_list.append({
+                "key" : current_node.key,
+                "value" : current_node.value,
+            })
+            self.inorder_helper(current_node.right, node_list)
+        return node_list
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def inorder(self):
-        pass
+        return self.inorder_helper(self.root)
+
+    def preorder_helper(self, current_node, node_list = None):
+        if node_list == None:
+            node_list = []
+        if current_node:
+            node_list.append({
+                "key" : current_node.key,
+                "value" : current_node.value,
+            })
+            self.preorder_helper(current_node.left, node_list)
+            self.preorder_helper(current_node.right, node_list)
+        return node_list
 
-    # Time Complexity: 
-    # Space Complexity:     
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)  
     def preorder(self):
-        pass
+        return self.preorder_helper(self.root)
 
-    # Time Complexity: 
-    # Space Complexity:     
+    def postorder_helper(self, current_node, node_list = None):
+        if node_list == None:
+            node_list = []
+        if current_node:
+            self.postorder_helper(current_node.left, node_list)
+            self.postorder_helper(current_node.right, node_list)
+            node_list.append({
+                "key" : current_node.key,
+                "value" : current_node.value,
+            })
+        return node_list
+
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)  
     def postorder(self):
-        pass
+        return self.postorder_helper(self.root)
 
-    # Time Complexity: 
-    # Space Complexity:     
-    def height(self):
-        pass
+    def height_helper(self, current_node, node_list = None):
+        if node_list == None:
+            node_list = []
+        if current_node:
+            self.height_helper(current_node.left, node_list)
+            if current_node.left == None and current_node.right == None:
+                node_list.append(current_node.key)
+            self.height_helper(current_node.right, node_list)
+        return node_list
+    
+    def find_count(self, key):
+        current_node = self.root
+        count = 0
+        while current_node:
+            count += 1
+            if key == current_node.key:
+                return count
+            elif key > current_node.key:
+                current_node = current_node.right
+            elif key < current_node.key:
+                current_node = current_node.left
+        return None
 
+    # Time Complexity: O(nlog(n))
+    # Space Complexity: O(m) where m is the number of end nodes  
```

Comment:
✨ This time and space complexity is correct for your solution!

You could refactor your code to achieve an O(n) time and space solution. Instead of having `height_helper` return a list, have it return an integer. When you make a recursive call on the left/right subtree, you know that your height is 1 + the height of the deepest subtree. So you can say that the height is 1 + max(height of left subtree, height of right subtree).